### PR TITLE
feat: add min gasPrice for gw_submit_l2Tx rpc

### PR DESCRIPTION
**Pull request by RetricSu**
dep: 

- https://github.com/nervosnetwork/godwoken-web3/pull/392

### CODE REVIEW

File: packages/api-server/src/methods/modules/gw.ts [hunk 1]
```diff
@@ -322,8 +329,51 @@ export class Gw {
    * @param args [l2tx(HexString)]
    * @returns
    */
-  async submit_l2transaction(args: any[]) {
+  async submit_l2transaction(args: [HexString]) {
     try {
+      // validate l2 tx params
+      const serialized2Tx = args[0];
```
Comment (classicalliu) [inline packages/api-server/src/methods/modules/gw.ts:335]:
```suggestion
      const serializedL2Tx = args[0];
```

File: packages/api-server/src/parse-tx.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    raw: DenormalizeRawL2Transaction(l2Tx.getRaw()),
+    signature: new Reader(l2Tx.getSignature().raw()).serializeJson(),
+  };
+}
+
+export function DenormalizeRawL2Transaction(rawL2Tx: schemas.RawL2Transaction) {
+  return {
+    chain_id: Uint64.fromLittleEndian(
+      new Reader(rawL2Tx.getChainId().raw()).serializeJson()
+    ).toHex(),
```
Comment (classicalliu) [inline packages/api-server/src/parse-tx.ts:72]:
```suggestion
    chain_id: new Uint64(
      rawL2Tx.getChainId().toLittleEndianBigUint64()
    ).toHex(),
```

**Review by Flouse — COMMENTED** [review]
Please rebase main branch.

File: packages/api-server/src/parse-tx.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  ).toHex();
+  const value = Uint128.fromLittleEndian(
+    "0x" + buf.slice(32, 48).toString("hex")
+  ).toHex();
+
+  const inputSize = Uint32.fromLittleEndian(
+    "0x" + buf.slice(48, 52).toString("hex")
+  );
+
+  const input = "0x" + buf.slice(52, 52 + inputSize.getValue()).toString("hex");
```
[the remark this answers is not in the document]
Comment (RetricSu) [reply packages/api-server/src/parse-tx.ts:42]: yes should check the size

File: packages/api-server/src/parse-tx.ts (hunk 2 above)
Comment (RetricSu) [reply packages/api-server/src/parse-tx.ts]: updated

**Comment by RetricSu** [comment]
> Please rebase main branch.

updated.

File: packages/api-server/src/parse-tx.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (RetricSu) [reply packages/api-server/src/parse-tx.ts:54]: updated

**Review by Flouse — APPROVED** [review]

**Review by classicalliu — APPROVED** [review]

**Pull request merged**